test/y2storage/filesystems/tmpfs_test.rb

```diff
@@ -24,24 +24,60 @@
 describe Y2Storage::Filesystems::Tmpfs do
   before do
     Y2Storage::StorageManager.create_test_instance
+
+    # Let's assume 8 GiB of RAM
+    allow(Yast::SCR).to receive(:Read).with(path(".proc.meminfo"))
+      .and_return("memtotal" => 8388608)
+
+    # And a block size of 4K
+    allow(Y2Storage::StorageManager.instance.arch).to receive(:page_size).and_return 4096
   end
+
   subject(:filesystem) { described_class.create(fake_devicegraph) }
 
   describe "#size" do
-    it "returns the size specified as mount option" do
-      filesystem.mount_path = "/tmp"
-      filesystem.mount_point.mount_options = ["rw", "size=128M", "noatime"]
-
-      expect(filesystem.size).to eq Y2Storage::DiskSize.MiB(128)
+    context "if there is no mount point (which should never happen)" do
+      it "returns half of the ram size (tmpfs default)" do
+        expect(filesystem.size).to eq Y2Storage::DiskSize.GiB(4)
+      end
     end
 
-    it "returns zero if there is no mount point" do
-      expect(filesystem.size).to eq Y2Storage::DiskSize.zero
-    end
+    context "if there is a mount point (which is the only supported case)" do
+      before { filesystem.mount_path = "/tmp" }
+
+      it "returns half of the ram size (tmpfs default) if there are no mount options" do
+        expect(filesystem.size).to eq Y2Storage::DiskSize.GiB(4)
+      end
+
+      it "returns the size limit directly specified as a size= mount option" do
+        filesystem.mount_point.mount_options = ["rw", "size=128M", "noatime"]
+
+        expect(filesystem.size).to eq Y2Storage::DiskSize.MiB(128)
+      end
+
+      it "returns the size limit specified as a percentage in the size= mount option" do
+        filesystem.mount_point.mount_options = ["rw", "size=25%"]
+
+        expect(filesystem.size).to eq Y2Storage::DiskSize.GiB(2)
+      end
+
+      it "returns zero if the size= mount option contains an invalid value" do
+        filesystem.mount_point.mount_options = ["size=big", "rw"]
+
+        expect(filesystem.size).to eq Y2Storage::DiskSize.zero
+      end
+
+      it "returns the size limit specified with the nr_blocks= mount option" do
+        filesystem.mount_point.mount_options = ["size=whatever", "nr_blocks=25k"]
+
+        expect(filesystem.size).to eq(Y2Storage::DiskSize.KiB(25) * 4096)
+      end
+
+      it "returns zero if the nr_blocks= mount option contains an invalid value" do
+        filesystem.mount_point.mount_options = ["nr_blocks=many"]
 
-    it "returns zero if there are no mount options" do
-      filesystem.mount_path = "/tmp"
-      expect(filesystem.size).to eq Y2Storage::DiskSize.zero
+        expect(filesystem.size).to eq Y2Storage::DiskSize.zero
+      end
```

Comment:
NP: I miss a test to reflect the precedence:

~~~
nr_blocks=512,size=500M
tmpfs (rw,relatime,size=512000k,inode64)

size=500M,nr_blocks=512
tmpfs (rw,relatime,size=2048k,inode64)
~~~